spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -191,6 +191,24 @@ private void publishConsumerFailedToStart() {
 		}
 	}
 
+	@Override
+	public void pause() {
+		super.pause();
+		DefaultPulsarMessageListenerContainer<T>.Listener consumer = this.listenerConsumer;
+		if (consumer != null) {
+			consumer.pause();
+		}
+	}
+
+	@Override
+	public void resume() {
+		super.resume();
+		DefaultPulsarMessageListenerContainer<T>.Listener consumer = this.listenerConsumer;
+		if (consumer != null) {
```

Comment:
I think we will want to resume the consumer before we call super.resume which signals to read from the consumer. As it is, I could imagine a race condition where the thread gets pre-empted at line 208 but the container resumes and tries to hit the consumer for data but its not yet resumed.